samcli/lib/cookiecutter/interactive_flow_creator.py

```diff
@@ -0,0 +1,125 @@
+    pass
+
+
+class InteractiveFlowCreator:
+    @staticmethod
+    def create_flow(flow_definition_path: str, extra_context: Optional[Dict] = None):
+        """
+        This method parses the given json/yaml file to create an InteractiveFLow. It expects the file to define
+        a list of questions. It parses the questions and add it to the flow in the same order they are defined
+        in the file, i.e. the default next-question of a given question will be the next one defined in the file,
+        while also respecting the question-defined _next_question map if provided.
+
+        Parameters:
+        ----------
+        flow_definition_path: str
+            A path to a json/yaml file that defines the questions of the flow. the file is expected to be in the
+            following format:
+            {
+                "questions": [
+                    {
+                      "key": "key of the corresponding cookiecutter config",
+                      "question": "the question to prompt to the user",
+                      "kind": "the kind of the question, for example: confirm",
+                      "isRequired": true/false,
+                      # optional branching logic to jump to a particular question based on the answer. if not given
+                      # will automatically go to next question
+                      "nextQuestion": {
+                        "True": "key of the question to jump to if the user answered 'Yes'",
+                        "False": "key of the question to jump to if the user answered 'Yes'",
+                      }
+                    },
+                    ...
+                ]
+            }
+        extra_context: Dict
+            if the template contains variable($variableName) this parameter provides the values for those variables.
+
+        Returns: InteractiveFlow(questions={k1: q1, k2: q2, ...}, first_question_key="first question's key")
+        """
+        questions, first_question_key = InteractiveFlowCreator._load_questions(flow_definition_path, extra_context)
+        return InteractiveFlow(questions=questions, first_question_key=first_question_key)
+
+    @staticmethod
+    def _load_questions(
+        flow_definition_path: str, extra_context: Optional[Dict] = None
+    ) -> Tuple[Dict[str, Question], str]:
+        previous_question: Optional[Question] = None
+        first_question_key: str = ""
+        questions: Dict[str, Question] = {}
+        questions_definition = InteractiveFlowCreator._parse_questions_definition(flow_definition_path, extra_context)
+
+        for question in questions_definition.get("questions"):
+            q = InteractiveFlowCreator._create_question_from_json(question)
+            if not first_question_key:
+                first_question_key = q.key
+            elif previous_question and not previous_question.default_next_question_key:
+                previous_question.set_default_next_question_key(q.key)
```

Comment:
I think it is better that the default next question should be defined in the questions file itself, as by this way, how can we differentiate if some question is the last question of the follow or not.

example:
```
[
     {
         "key": "q1",
         "nextQuestion": {
                  "True": "q2",
                  "False": "q3",
         }
     },
     {
         "key": "q2"
     },
     {
         "key": "q3"
     }
]
```
my assumption in the above questions is q2 or q3 are the end of the interactive flow, but using the code above I think you will add q3 as the default new question of q2

Reply:
by default a next question for a question is the one defined after it in the questions' array. We don't need to define this`nextQuestion` map unless we have the need to do branching. I understand that this will need a last question that all branches terminates to but this is usually the case in the interactive-flows, you end the questionnaire with a message. For example, in the case you provided it will be something like this:

```
[
     {
         "key": "q1",
         "nextQuestion": {
                  "True": "q2",
                  "False": "q3",
         }
     },
     {
         "key": "q2"
         "defaultNextQuestion": "q4"
     },
     {
         "key": "q3"
     },
     {
         "key": "q4"
         "kind": "info"
         "question": "End of Text 👯👯 "
     }
]
```